### PR TITLE
James 1697 JPA fields should not be final

### PR DESCRIPTION
**Pull request by chibenwa**
This ticket solve issue with JPA backend

It also enables MPT tests for JPA

And it uses features for MPT IMAP tests everywhere.

### CODE REVIEW

File: mpt/core/src/main/java/org/apache/james/mpt/api/ImapFeatures.java [hunk 1]
```diff
@@ -28,7 +28,10 @@
 public class ImapFeatures {
     
     public enum Feature {
-        NAMESPACE_SUPPORT
```
Comment (aduprat) [inline mpt/core/src/main/java/org/apache/james/mpt/api/ImapFeatures.java]: We may define this in a _Capabilites_ class in james-mailbox-api project, and use it everywhere.

Comment (chibenwa) [inline mpt/core/src/main/java/org/apache/james/mpt/api/ImapFeatures.java]: Ok, you mean use the newly introduced Mailbox capabilities for this ?

Seems nice, I will introduce this commit on monday.

**Comment by mbaechler** [comment]
:+1:

File: mpt/core/src/main/java/org/apache/james/mpt/api/ImapFeatures.java (hunk 1 above)
Comment (chibenwa) [inline mpt/core/src/main/java/org/apache/james/mpt/api/ImapFeatures.java]: Antoine, I have a commit for your last request, but I don't like it : 
 - I encounter difficulties with Quota support, which depends on wether Quota stuff was setted or not (mutable by definition) while trying to do something immutable (move ImapFeatures to mailbox/api, renamed MailboxFeature)
 - I have one more concern : for Cyrus, ImapFeature is more suited than MailboxFeature

**Comment by chibenwa** [comment]
Can I merge this ?

**Comment by mbaechler** [comment]
Yes

**Pull request closed without merging**